Sourcing/DataProviderSwitcher.swift

```diff
@@ -22,6 +22,7 @@
 
 import Foundation
 
+/// `DataProviderSwitcher` lets one switch dynamicly between different DataProviders by changing corresponing to a given state.
```

Comment:
`DataProviderSwitcher` allows switching between different DataProviders dynamically by changing to a given state.